lib/src/public_interface.dart

```diff
@@ -0,0 +1,166 @@
+import 'package:yaml/yaml.dart';
+
+import './source_edit.dart';
+
+/// An interface for modififying [YAML][1] documents while preserving comments and
+/// whitespaces.
+///
+/// YAML parsing is supported by `package:yaml`, and modifications are performed as
+/// string operations. Each time a modification takes place via one of the public
+/// methods, we calculate the expected final result, and parse the result YAML string,
+/// and ensure the two YAML trees match, throwing an exception otherwise. Such a situation
+/// should be extremely rare, and should only occur with degenerate formatting.
+///
+/// Users may define the default settings to be applied to these string modifications.
+/// Note however that these settings only apply to portions of the YAML that are modified
+/// by this class.
+///
+/// Most modification methods require the user to pass in an [Iterable<Object>] path that
+/// holds the keys/indices to navigate to the element. Key equality is performed via
+/// `package:yaml`'s [YamlMap]'s key equality.
+///
+/// [1]: https://yaml.org/
+abstract class YamlEditor {
+  /// The configuration to be used for the various string manipulation operations.
+  final YamlStyle config;
+
+  /// List of [SourceEdit]s that have been applied to [_yaml] since the creation of this
+  /// instance, in chronological order. Intended to be compatible with `package:analysis_server`.
+  final List<SourceEdit> edits = [];
```

Comment:
Let's make this a getter that returns an [UnmodifiableListView](https://api.dart.dev/stable/2.8.4/dart-collection/UnmodifiableListView-class.html).. And let's return a view of a copy of the internal list..
Such that updating the internal list does not update the list returned here.

Reply:
Done!